server/configuration-validation.js

```diff
@@ -1,125 +1,192 @@
 const { isObject } = require('./utils')
 
-/**
- * Validates the plugin configuration provided in `plugins/config.js` of the users plugin configuration.
- * Modifies the value of config on place.
- *
- * @param  {object} config - configurations
- */
-function validateConfiguration(config) {
-  if (config === undefined) {
-    return
-  }
+function CollectionOptions(name, configuration) {
```

Comment:
I think it's a bit confusing that you named everything `configuration` in this file. You might want to use `collection` for example, and use `validateCollection` as the object validator?